src/modules/api/transaction/dtos/index.ts

```diff
@@ -121,20 +119,14 @@ export class AdminTransactionHistoryDto {
 
     @IsOptional()
     @IsString()
-    transactionId: string;
-
-    @IsOptional()
-    @IsPhoneNumber("NG")
-    @Length(11, 11, {
-        message: "Phone number must be valid containing 11 digits",
-    })
-    phone: string;
-
-    @IsOptional()
-    @IsString()
-    meterNo: string;
+    referenceId: string;
```

Comment:
Use a generic name instead. change this to searchName. Be always logical with your variable naming.